examples/ScanLoop.cpp

```diff
@@ -0,0 +1,65 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// Copyright 2020 Lawrence Livermore National Security, LLC and other CARE developers.
+// See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: BSD-3-Clause
+//////////////////////////////////////////////////////////////////////////////////////
+
+// This macro needs to be defined before including care/care.h,
+// which allows you to port a file at a time. Without this define,
+// LOOP_STREAM will run on the CPU. With this define, LOOP_STREAM
+// will run on the GPU only if you built with CUDA enabled. Otherwise,
+// if CUDA is disabled, it will default back to running on the host.
+#define GPU_ACTIVE
+
+// CARE library header
+#include "care/care.h"
+#include "care/scan.h"
+
+// Std library headers
+#include <iostream>
+#include <vector>
+
+int main(int, char**) {
+   // Normal code
+   int size = 1000;
+   std::vector<int> data;
+
+   int threshold = 420;
+
+   for (int i = 0; i < size; ++i) {
+      int value = i * i;
+
+      if (value < threshold) { 
+         data.push_back(value);
+      }
+   }
+
+   std::cout << "CPU count: " << data.size() << std::endl;
+
+   // Ported code
+   int size2 = 1000;
+   care::host_device_ptr<int> data2(size2);
+
+   int count = 0;
+   int threshold2 = 420;
+
+   SCAN_LOOP(i, 0, size2, pos, count, i * i < threshold2) {
+      data2[pos] = i * i;
```

Comment:
Comment on each argument of the SCAN LOOP, what the purpose is. Since this is under examples, explain even basic things thoroughly

Reply:
Provided the equivalent code so they can understand what the macro is doing.